src/components/bcros/comment/Index.vue

```diff
@@ -0,0 +1,144 @@
+<script setup lang="ts">
+import type { CommentIF } from '@bcrs-shared-components/interfaces'
+import { createComment } from '~/utils/filings'
+
+const t = useNuxtApp().$i18n.t
+const noChangesSinceSave = ref(false)
+
+const props = withDefaults(defineProps<{
+    comments?: Array<CommentIF>,
+    showCloseModal?: boolean,
+    filing: ApiResponseFilingI
+  }>(), {
+  comments: () => [],
+  showCloseModal: true
+})
+
+defineEmits(['close'])
+
+const MAX_COMMENT_LENGTH = 2000
+const commentToAdd = ref('')
+
+watch(commentToAdd, (newComment, oldComment) => {
+  if (newComment !== oldComment) {
+    noChangesSinceSave.value = false
+  }
+})
+const error = computed((): string => {
+  if (!commentToAdd || !commentToAdd.value || commentToAdd.value.length === 0) {
+    if (noChangesSinceSave.value === true) {
+      return ''
+    }
+    return t('label.comments.commentRequired')
+  }
+  // if (commentToAdd && commentToAdd.value && commentToAdd.value.length > MAX_COMMENT_LENGTH) {
+  //   return t('label.comments.commentTooLong', MAX_COMMENT_LENGTH)
+  // }
+  return ''
+})
+
+const textAreaUi = {
+  rounded: '',
+  variant: {
+    bottom:
+      'bg-transparent border-b-2 border-b-bcGovGray-600 focus:border-b-bcGovGray-600 focus:ring-0 focus:shadow-none',
+    bottomError:
+      'bg-transparent border-b-2 border-b-red-600 focus:border-b-red-600 focus:ring-0 focus:shadow-none'
+  }
+}
+
+const saveComment = async () => {
+  if (error.value !== '') {
+    return
+  }
+  if (commentToAdd.value.length > MAX_COMMENT_LENGTH) {
+    return
+  }
+
+  if (commentToAdd.value.length === 0) {
+    return
+  }
+
+  try {
+    await createComment(props.filing, commentToAdd.value)
+  } catch (e) {
+    error.value = `Error saving comment: ${e.message}`
+  }
+
+  commentToAdd.value = ''
+  noChangesSinceSave.value = true
+}
+
+</script>
+
+<template>
+  <!-- comment-text-->
+  <div class="px-5 py-5">
+    <slot name="comment-header">
+      <span class="text-primary-600">
+        <UIcon name="i-mdi-comment-text" class="w-5 h-5" />
+        {{ $t('label.comments.comment', (comments?.length || 0 )) }}
+      </span>
+      <span v-if="showCloseModal" class="float-right">
+        <UIcon name="i-mdi-close" class="w-5 h-5 cursor-pointer" @click="$emit('close')" />
+      </span>
+    </slot>
+
+    <slot name="comment-add-area">
+      <div class="mb-6" data-cy="comment-add-slot">
+        <UTextarea
+          v-model="commentToAdd"
+          data-cy="comment-add-textarea"
+          :variant="error ? 'bottomError' : 'bottom'"
+          :rows="5"
+          :ui="textAreaUi"
+        />
+        <p v-if="error" class="text-red-600">
```

Comment:
if you wrap the UTextarea with a UFormGroup then you can remove all these <p> blocks and the styling for text colour will come from the our common file, which would be better I think. You would have to change the 'charactersExceeded' message to come from your error message logic

`<UFormGroup :error="error" :help="MAX_COMMENT_LENGTH - commentToAdd.length">...`

Reply:
done